ArduCopter/AP_Arming.cpp

```diff
@@ -659,11 +657,11 @@ bool AP_Arming_Copter::mandatory_checks(bool display_failure)
     AP_Notify::flags.pre_arm_gps_check = result;
 
     // call mandatory alt check
-    if (!alt_checks(display_failure)) {
+    if (result && !alt_checks(display_failure)) {
         result = false;
     }
 
-    return result & AP_Arming::mandatory_checks(display_failure);
+    return result && AP_Arming::mandatory_checks(display_failure);
```

Comment:
this change means that we will not call all checks, lazy evaluation will skip `alt_checks` and `AP_Arming::mandatory_checks` if result is false. That means we don't get all the info on the first attempt.